TangoRosStreamer/tango_ros_node/src/main/java/eu/intermodalics/tango_ros_node/TangoRosNode.java

```diff
@@ -70,12 +80,12 @@ public void attachCallbackListener(CallbackListener callbackListener) {
     @Override
     public void onError(Node node, Throwable throwable) {
         super.onError(node, throwable);
-        if (throwable.getMessage().contains(ROS_CONNECTION_FAILURE_ERROR_MSG) ||
-                throwable.getMessage().contains(ROS_CONNECTION_UNREACHABLE_ERROR_MSG) ||
-                throwable.getMessage().contains(ROS_CONNECTION_TIMEOUT_ERROR_MSG) ||
-                throwable.getMessage().contains(ROS_WRONG_HOST_NAME_ERROR_MSG)) {
-            executeOnErrorHook(ROS_CONNECTION_ERROR);
-        } else if (this.executeReturnCode != 0) {
+        for (String errorMessage : errorMessages) {
+            if (throwable.getMessage().contains(errorMessage)) {
+                executeOnErrorHook(ROS_CONNECTION_ERROR);
+            }
```

Comment:
nit: perhaps I would add a a `return` statement as the error hook should be executed only once, so there's no need to keep iterating.